packages/flutter/lib/src/material/carousel.dart

```diff
@@ -252,8 +351,8 @@ class _CarouselViewState extends State<CarouselView> {
               offset: position,
               clipBehavior: Clip.antiAlias,
               slivers: <Widget>[
-                _SliverFixedExtentCarousel(
-                  itemExtent: _itemExtent,
+                if (_itemExtent != null) _SliverFixedExtentCarousel(
```

Comment:
Rather than inlining the choice of sliver with if statements, can you create a private method that returns the right sliver here for better readability?